janus_server/src/trace.rs

```diff
@@ -109,6 +139,82 @@ pub fn install_trace_subscriber(config: &TraceConfiguration) -> Result<(), Error
         );
     }
 
+    if config.otel_jaeger && config.otel_otlp.enabled {
+        return Err(Error::Other(anyhow::anyhow!(
+            "bad configuration, both Jaeger and OTLP OpenTelemetry layers cannot be enabled at the same time"
+        )));
+    }
+
+    #[cfg(feature = "jaeger")]
+    if config.otel_jaeger {
+        let tracer = opentelemetry_jaeger::new_pipeline()
+            .with_service_name("janus_server")
+            .install_batch(opentelemetry::runtime::Tokio)?;
+        let telemetry = tracing_opentelemetry::layer().with_tracer(tracer);
+        layers.push(telemetry.boxed());
+    }
+
+    #[cfg(not(feature = "jaeger"))]
+    if config.otel_jaeger {
+        eprintln!(
+            "Warning: the OpenTelemetry Jaeger subscriber was enabled in the \
+            configuration file, but support was not enabled at compile time. \
+            Rebuild with `--features jaeger`."
+        );
+    }
+
+    #[cfg(feature = "otlp")]
+    if config.otel_otlp.enabled {
+        use opentelemetry::{
+            sdk::{trace, Resource},
+            KeyValue,
+        };
+        use opentelemetry_otlp::WithExportConfig;
+        use opentelemetry_semantic_conventions::resource::SERVICE_NAME;
+        use std::str::FromStr;
+        use tonic::metadata::{MetadataKey, MetadataMap, MetadataValue};
+        use tracing_subscriber::filter::{LevelFilter, Targets};
+
+        let mut map = MetadataMap::with_capacity(config.otel_otlp.metadata.len());
+        for (key, value) in config.otel_otlp.metadata.iter() {
+            map.insert(MetadataKey::from_str(key)?, MetadataValue::from_str(value)?);
+        }
+
+        let tracer =
+            opentelemetry_otlp::new_pipeline()
+                .tracing()
+                .with_exporter(
+                    opentelemetry_otlp::new_exporter()
+                        .tonic()
+                        .with_endpoint(config.otel_otlp.endpoint.clone())
+                        .with_metadata(map),
+                )
+                .with_trace_config(trace::config().with_resource(Resource::new(vec![
+                    KeyValue::new(SERVICE_NAME, "janus_server"),
+                ])))
+                .install_batch(opentelemetry::runtime::Tokio)?;
+
+        let filter = Targets::new()
+            .with_default(LevelFilter::TRACE)
+            .with_target("h2::proto::streams::prioritize", LevelFilter::OFF)
+            .with_target("h2::proto::streams::store", LevelFilter::OFF)
+            .with_target("h2::codec::framed_write", LevelFilter::OFF);
```

Comment:
I'm sure these are sensible choices but a comment explaining why we are enabling (or disabling?) these features would be appreciated.